src/frontend/http.rs

```diff
@@ -562,6 +569,8 @@ mod tests {
     }
 
     const SELECT_QUERY: &str = "SELECT COUNT(*) AS c FROM test_table";
+    const PERCENT_ENCODED_SELECT_QUERY: &str = "SELECT%20COUNT(*)%20AS%20c%20FROM%20test_table";
```

Comment:
Can you also add a case here for partial urlencoding? I'm thinking something like 

```sql
SELECT COUNT(*) AS "%F0%9F%99%82" FROM test_table
```

to test that this will still behave if we let users save space in the HTTP header by only urlencoding nonprintable/nonASCII characters

Reply:
We already kind of have that since `PERCENT_ENCODED_SELECT_QUERY` is the same as `SELECT_QUERY` with spaces replaced by `%20`. The hash of the two queries is even the same (since hash is computed after decoding).

Reply:
Right, but I meant we should test that a space (that isn't urlencoded) still remains a space after urlencoding. But I'm now realizing that the rest of the test suite that exercises this code does that anyway (no urlencoding), so we don't need to do this.